components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4907,11 +4907,15 @@ public static String resolveExternalConsentPageUrl(String tenantDomain) throws I
      */
     public static boolean isFapiConformantApp(String clientId) throws IdentityOAuth2Exception {
 
-        ServiceProvider serviceProvider = getServiceProvider(clientId);
-        ServiceProviderProperty[] serviceProviderProperties = serviceProvider.getSpProperties();
-        for (ServiceProviderProperty serviceProviderProperty : serviceProviderProperties) {
-            if (IS_FAPI_CONFORMANT_APP.equals(serviceProviderProperty.getName())) {
-                return Boolean.parseBoolean(serviceProviderProperty.getValue());
+        boolean enableFAPIValidation = IdentityUtil.getProperty(OAuthConstants.ENABLE_FAPI) != null ?
+                Boolean.parseBoolean(IdentityUtil.getProperty(OAuthConstants.ENABLE_FAPI)) : false;
+        if (enableFAPIValidation) {
```

Comment:
lets check !enableFAPIValidation and return false without nesting the if condition

Reply:
fixed